examples_tests/23.Autoexposure/main.cpp

```diff
@@ -4,20 +4,771 @@
+			auto outImage = logicalDevice->createDeviceLocalGPUImageOnDedMem(std::move(creationParams));
+
+			// transition the image to GENERAL
+			{
+				const auto& imageCreationParams = outImage->getCreationParameters();
+
+				const auto& cb = cmdbufs[0];
+
+				video::IGPUCommandBuffer::SImageMemoryBarrier toGeneral = {};
+				toGeneral.image = outImage;
+				toGeneral.oldLayout = asset::EIL_UNDEFINED;
+				toGeneral.newLayout = asset::EIL_GENERAL;
+				toGeneral.subresourceRange.aspectMask = video::IGPUImage::EAF_COLOR_BIT;
+				toGeneral.subresourceRange.baseArrayLayer = 0u;
+				toGeneral.subresourceRange.layerCount = imageCreationParams.arrayLayers;
+				toGeneral.subresourceRange.baseMipLevel = 0u;
+				toGeneral.subresourceRange.levelCount = imageCreationParams.mipLevels;
+
+				cb->begin(video::IGPUCommandBuffer::EU_ONE_TIME_SUBMIT_BIT);
+				cb->pipelineBarrier(
+					asset::EPSF_TOP_OF_PIPE_BIT,
+					asset::EPSF_BOTTOM_OF_PIPE_BIT,
+					asset::EDF_BY_REGION_BIT,
+					0u, nullptr,
+					0u, nullptr,
+					1u, &toGeneral);
+				cb->end();
+
+				auto fence = logicalDevice->createFence(video::IGPUFence::ECF_UNSIGNALED);
+
+				video::IGPUQueue::SSubmitInfo submit;
+				submit.commandBufferCount = 1u;
+				submit.commandBuffers = &cb.get();
+				graphicsComputeQueue->submit(1u, &submit, fence.get());
+
+				logicalDevice->blockForFences(1u, &fence.get());
+			}
+
+			auto viewCreationParams = imageToTonemapView->getCreationParameters();
+			viewCreationParams.format = outImage->getCreationParameters().format;
+			viewCreationParams.image = outImage;
+			tonemapperOutImageViews[i] = logicalDevice->createGPUImageView(std::move(viewCreationParams));
+		}
+
+		const float meteringMinUV[2] = { 0.1f,0.1f };
+		const float meteringMaxUV[2] = { 0.9f,0.9f };
+		LumaMeterClass::Uniforms_t<MeterMode> uniforms;
+		lumaDispatchInfo = LumaMeterClass::buildParameters(
+			uniforms,
+			imageToTonemapView->getCreationParameters().image->getCreationParameters().extent,
+			meteringMinUV, meteringMaxUV);
+
+
+		// create luma meter ubo
+		{
+			const size_t bufferSize = sizeof(uniforms);
+			video::IGPUBuffer::SCreationParams creationParams = {};
+			creationParams.usage = static_cast<video::IGPUBuffer::E_USAGE_FLAGS>(video::IGPUBuffer::EUF_UNIFORM_BUFFER_BIT | video::IGPUBuffer::EUF_TRANSFER_DST_BIT);
+			lumaMeterUbo = logicalDevice->createDeviceLocalGPUBufferOnDedMem(creationParams, bufferSize);
+
+			// fill it up with `uniforms`
+			core::smart_refctd_ptr<video::IGPUFence> fence = logicalDevice->createFence(video::IGPUFence::ECF_UNSIGNALED);
+			asset::SBufferRange<video::IGPUBuffer> bufferRange;
+			bufferRange.offset = 0ull;
+			bufferRange.size = lumaMeterUbo->getCachedCreationParams().declaredSize;
+			bufferRange.buffer = lumaMeterUbo;
+			utilities->updateBufferRangeViaStagingBuffer(
+				fence.get(),
+				queues[CommonAPI::InitOutput::EQT_TRANSFER_UP],
+				bufferRange,
+				&uniforms);
+			logicalDevice->blockForFences(1u, &fence.get());
+		}
```

Comment:
@Erfan-Ahmadi after this gets merged to `master` and then `master` is merged into NewAPI, turn this into a 1-liner

the reason why it isnt a 1 liner cause the API sucks (no usage flags in the specialized methods or `IUtilities::createFilled`)